docs/source/platform/federation.mdx

```diff
@@ -0,0 +1,354 @@
+
+1. The partial schema is uploaded to a secure location and indexed
+1. The implementing service is updated in the registry to point to the partial schema
+1. All implementing services are composed in the cloud to produce a new complete schema
+1. If composition fails, the command exits and emits errors
+1. If composition succeeds, the top-level managed configuration file is updated in-place to point to the updated set of implementing services
+
+On the other side of the equation, sits the gateway. The gateway is constantly listening for changes to the top-level managed configuration file. The location of the managed configuration file is guarded by using a hash of the API key, provisioned ahead of time so as not to affect reliability. The life-cycle of dynamic configuration updates is as follows:
+
+1. The gateway listens for updates to its managed configuration every 10 seconds
+1. On update, the gateway downloads configuration for each implementing service in parallel
+1. The gateway updates its query-planner to use the new configuration
+1. The gateway flushes its `queryPlan` cache to begin planning new queries with the new config
+1. In-flight requests finish being served and delivered to consumers as incoming operations are served as well
+
+### Integrating with CI / CD
+
+### Reliability and security
+
+The managed configuration for the Apollo gateway is exposed through [Google Cloud Storage](https://cloud.google.com/storage). For all API keys, the Apollo Graph Manager provisions a public file accessible via the hash of the API key used to gain access to the managed configuration files for the graph to which the API key belongs. In the event that managed configuration is inaccessible, due to an outage in Google's Cloud Storage service, the gateway will continue to serve the last-known configuration. In the event that that Apollo Graph Manager API is down, changes to managed configuration will be stalled, but the last-published managed configuration files will still be accessible via GCS.
+
+### Using variants to control rollout
+
+With [managed federation](#managed-federation), you have the ability to control which version of your graph a fleet of Gateways are running with. For the majority of deployments, rolling over all of your Gateways to a new schema version is a good strategy, since changes should be checked to be backwards compatible using [Schema Validation](/platform/schema-validation/). However, changes at the gateway level may involve a variety of different updates, like changing how query plans are generated or transferring type ownership from one service to another. In the case that your infrastructure requires more advanced deployment strategies, we recommend using [graph variants](/platform/schema-registry/#registering-schemas-to-a-variant) to manage different fleets of Gateways running with different configurations.
```

Comment:
```suggestion
With [managed federation](#managed-federation), you have the ability to control which version of your graph a fleet of gateways are running with. For the majority of deployments, rolling over all of your gateways to a new schema version is a good strategy, since changes should be checked to be backwards compatible using [Schema Validation](/platform/schema-validation/). However, changes at the gateway level may involve a variety of different updates, like transferring type ownership from one service to another. In the case that your infrastructure requires more advanced deployment strategies, we recommend using [graph variants](/platform/schema-registry/#registering-schemas-to-a-variant) to manage different fleets of gateways running with different configurations.
```
Gateway(s) -> gateway(s)

Removed - technically correct, but maybe unnecessary and seems to imply that this is configurable
>changing how query plans are generated or